src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,876 @@
+    return CachedBasis(basisType, basis)
+end
+
+#=
+get_coordinates(::Tucker, A, X::TuckerTVector, b)
+
+The coordinates of a tangent vector X at point A on the Tucker manifold with respect to the
+basis b.
+=#
+function get_coordinates(::Tucker, 𝔄, X::TuckerTVector, ℬ::CachedHOSVDBasis)
+    coords = vec(X.Ċ)
+    for d in 1:length(X.U̇)
+        coord_mtx = (ℬ.data.U⊥[d] \ X.U̇[d]) * Diagonal(𝔄.hosvd.σ[d])
+        coords = vcat(coords, vec(coord_mtx'))
+    end
+    return coords
+end
+function get_coordinates(
+    M::Tucker,
+    𝔄,
+    X,
+    ℬ::DefaultOrthonormalBasis{𝔽,TangentSpaceType},
+) where {𝔽}
+    return get_coordinates(M, 𝔄, X, get_basis(M, 𝔄, ℬ))
+end
+
+#=
+get_vector(::Tucker, A, x, b)
+
+The tangent vector at a point A whose coordinates with respect to the basis b are x.
+=#
+function get_vector!(
+    ::Tucker,
+    y,
+    𝔄::TuckerPoint,
+    x::AbstractVector{T},
+    ℬ::CachedHOSVDBasis,
+) where {T}
+    ξ = convert(Vector{promote_type(number_eltype(𝔄), eltype(x))}, x)
+    ℭ = 𝔄.hosvd.core
+    σ = 𝔄.hosvd.σ
+    U⊥ = ℬ.data.U⊥
+    D = ndims(ℭ)
+    r⃗ = size(ℭ)
+    n⃗ = size(𝔄)
+
+    # split ξ into ξ_core and ξU so that vcat(ξ_core, ξU...) == ξ, but avoid copying
+    ξ_core = view(ξ, 1:length(ℭ))
+    ξU = Vector{typeof(ξ_core)}(undef, D)
+    nextcolumn = length(ℭ) + 1
+    for d in 1:D
+        numcols = r⃗[d] * (n⃗[d] - r⃗[d])
+        ξU[d] = view(ξ, nextcolumn:(nextcolumn + numcols - 1))
+        nextcolumn += numcols
+    end
+
+    # Construct ∂U[d] by plugging in the definition of the orthonormal basis [Dewaele2021]
+    # ∂U[d] = ∑ᵢⱼ { ξU[d]ᵢⱼ (σ[d]ⱼ)⁻¹ U⊥[d] 𝐞ᵢ 𝐞ⱼᵀ }
+    #       = U⊥[d] * ∑ⱼ (σ[d]ⱼ)⁻¹ (∑ᵢ ξU[d]ᵢⱼ  𝐞ᵢ) 𝐞ⱼᵀ
+    # ξU[d] = [ξ₁₁, ..., ξ₁ⱼ, ..., ξᵢ₁, ..., ξᵢⱼ, ..., ]
+    # => turn these i and j into matrix indices and do matrix operations
+    for d in 1:D
+        grid = transpose(reshape(ξU[d], r⃗[d], n⃗[d] - r⃗[d]))
+        mul!(y.U̇[d], U⊥[d], grid * Diagonal(1 ./ σ[d]))
+    end
+
+    y.Ċ .= reshape(ξ_core, size(y.Ċ))
+    return y
+end
+function get_vector!(
+    ℳ::Tucker,
+    y,
+    𝔄::TuckerPoint,
+    x,
+    ℬ::DefaultOrthonormalBasis{𝔽,TangentSpaceType},
+) where {𝔽}
+    return get_vector!(ℳ, y, 𝔄, x, get_basis(ℳ, 𝔄, ℬ))
+end
+
+function get_vectors(ℳ::Tucker, 𝔄::TuckerPoint{T,D}, ℬ::CachedHOSVDBasis) where {T,D}
+    vectors = Vector{TuckerTVector{T,D}}(undef, manifold_dimension(ℳ))
+    foreach((i, vᵢ) -> setindex!(vectors, copy(vᵢ), i), ℳ, 𝔄, ℬ)
+    return vectors
+end
+function get_vectors(ℳ::Tucker, 𝔄::TuckerPoint, ℬ::DefaultOrthonormalBasis)
+    return get_vectors(ℳ, 𝔄, get_basis(ℳ, 𝔄, ℬ))
+end
+
+"""
+    inner(M::Tucker, A::TuckerPoint, x::TuckerTVector, y::TuckerTVector)
+
+The Euclidean inner product between tangent vectors `x` and `y` at the point `A` on
+the Tucker manifold. This is equal to `embed(M, A, x) ⋅ embed(M, A, y)`.
+
+    inner(::Tucker, A::TuckerPoint, x::TuckerTVector, y)
+    inner(::Tucker, A::TuckerPoint, x, y::TuckerTVector)
+
```

Comment:
We usually use `p` for points on a manifold `M` and `X` and `Y` for the tangent vectors. I think you used a mathcal M instead often for the manifold.

If you use a different notation, please use it consistently; otherwise it would also be great if you could adapt to the notation we use (but you doing not have to); see https://juliamanifolds.github.io/Manifolds.jl/latest/misc/notation.html